src/core/public/chrome/nav_group/nav_group_service.ts

```diff
@@ -0,0 +1,136 @@
+   */
+  parentNavLinkId?: string;
+}
+
+export type NavGroupItemInMap = ChromeNavGroup & {
+  navLinks: ChromeRegistrationNavLink[];
+};
+
+export interface ChromeNavGroupServiceSetupContract {
+  addNavLinksToGroup: (navGroup: ChromeNavGroup, navLinks: ChromeRegistrationNavLink[]) => void;
+  /**
+   * Get a boolean value to indicates whether use case is enabled
+   */
+  getNavGroupEnabled: () => boolean;
+}
+
+export interface ChromeNavGroupServiceStartContract {
+  getNavGroupsMap$: () => Observable<Record<string, NavGroupItemInMap>>;
+  getNavGroupEnabled: ChromeNavGroupServiceSetupContract['getNavGroupEnabled'];
+}
+
+/** @internal */
+export class ChromeNavGroupService {
+  private readonly navGroupsMap$ = new BehaviorSubject<Record<string, NavGroupItemInMap>>({});
+  private readonly stop$ = new ReplaySubject(1);
+  private navLinks$: Observable<Array<Readonly<ChromeNavLink>>> = new BehaviorSubject([]);
+  private navGroupEnabled: boolean = false;
+  private navGroupEnabledUiSettingsSubscription: Subscription | undefined;
+  private addNavLinkToGroup(
+    currentGroupsMap: Record<string, NavGroupItemInMap>,
+    navGroup: ChromeNavGroup,
+    navLink: ChromeRegistrationNavLink
+  ) {
+    const matchedGroup = currentGroupsMap[navGroup.id];
+    if (matchedGroup) {
+      const links = matchedGroup.navLinks;
+      const isLinkExistInGroup = !!links.find((link) => link.id === navLink.id);
+      if (isLinkExistInGroup) {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `[ChromeService] Navlink of ${navLink.id} has already been registered in group ${navGroup.id}`
+        );
+        return currentGroupsMap;
+      }
+      matchedGroup.navLinks.push(navLink);
+    } else {
+      currentGroupsMap[navGroup.id] = {
+        ...navGroup,
+        navLinks: [navLink],
+      };
+    }
+
+    return currentGroupsMap;
+  }
+  private getSortedNavGroupsMap() {
+    return combineLatest([this.navGroupsMap$, this.navLinks$])
+      .pipe(takeUntil(this.stop$))
+      .pipe(
+        map(([navGroupsMap, navLinks]) => {
+          return Object.keys(navGroupsMap).reduce((sortedNavGroupsMap, navGroupId) => {
+            const navGroup = navGroupsMap[navGroupId];
+            const sortedNavLinks = getOrderedLinksOrCategories(
+              fulfillRegistrationLinksToChromeNavLinks(navGroup.navLinks, navLinks)
+            );
+            sortedNavGroupsMap[navGroupId] = {
+              ...navGroup,
+              navLinks: flattenLinksOrCategories(sortedNavLinks),
+            };
+            return sortedNavGroupsMap;
+          }, {} as Record<string, NavGroupItemInMap>);
+        })
+      );
+  }
+  setup({ uiSettings }: { uiSettings: IUiSettingsClient }): ChromeNavGroupServiceSetupContract {
+    this.navGroupEnabledUiSettingsSubscription = uiSettings
+      .get$('home:useNewHomePage', false)
+      .subscribe((value) => {
+        this.navGroupEnabled = value;
```

Comment:
Shall we add `requiresPageReload: true,` in `src/plugins/home/server/ui_settings.ts`? Then the page will show a toast `One or more settings require you to reload the page to take effect.`. The `navGroupEnabled` is a private property in `NavGroupService`, I think it can't be used outside directly.

Reply:
Agree, let's add requiresPageReload to the setting.